src/main/java/com/uid2/shared/attest/AttestationTokenRetriever.java

```diff
@@ -107,14 +107,18 @@ private void attestationExpirationCheck(long timerId) {
 
             if (currentTime.isAfter(tenMinutesBeforeExpire)) {
                 LOGGER.info("Attestation token is 10 mins from the expiry timestamp {}. Re-attest...", attestationTokenExpiresAt);
-                try {
-                    this.isAttesting = true;
-                    attest();
-                } catch (AttestationTokenRetrieverException | IOException e) {
-                    notifyResponseStatusWatcher(401);
-                    LOGGER.info("Re-attest failed: ", e);
-                } finally {
-                    this.isAttesting = false;
+                if (!attestationProvider.isReady()) {
```

Comment:
I would prefer return early pattern, instead of if/else.
```
if(abc) return;
xyz;
```

Help us reduce nested level.

Reply:
Sure, tried to follow the same style, let me update

Reply:
Updated.